### PR TITLE
[Feature] Add array function of array_repeat

### LINKED ISSUE #19658: [function] array_repeat
- [x] https://github.com/StarRocks/starrocks/pull/47862
need a function to repeatedly generate array.

## description

```
array_repeat (element, count)
```
in which,
element: can be any data type that supported in an array, including array.
count: integer

## examples

```
select array_repeat(1,5) ;
[1,1,1,1,1]

select  array_repeat([1,2],3);
[[1,2],[1,2],[1,2]]

```

### PR DESCRIPTION
**Pull request by chenminghua8**
## What I'm doing:

Fixes #19658

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by chenminghua8** [comment]
Can someone review this RP?

File: be/src/exprs/array_functions.cpp [hunk 1]
```diff
@@ -1627,4 +1627,65 @@ StatusOr<ColumnPtr> ArrayFunctions::array_sortby_multi(FunctionContext* ctx, con
...
+    size_t total_repeated_rows = 0;
+    for (int cur_row = 0; cur_row < num_rows; cur_row++) {
+        if (repeat_count_viewer.is_null(cur_row)) {
+            dest_offsets->append(total_repeated_rows);
+        } else {
+            Datum source_value = src_column->get(cur_row);
+            auto repeat_count = repeat_count_viewer.value(cur_row);
+            if (repeat_count > 0) {
+                for (int repeat_index = 0; repeat_index < repeat_count; repeat_index++) {
+                    TRY_CATCH_BAD_ALLOC(dest_column_elements->append_datum(source_value));
```
Comment (mofeiatwork) [inline be/src/exprs/array_functions.cpp:1660]: Use the `append(datum, N)` would be more efficient, which can eliminate the function call overhead

Comment (chenminghua8) [reply be/src/exprs/array_functions.cpp:1660]: ok

Comment (chenminghua8) [reply be/src/exprs/array_functions.cpp:1660]: @mofeiatwork I did not find a method signature similar to append(const Datum& datum, size_t count) in the Column class, but only append(const Column& src, size_t offset, size_t count), append(const Column& src) and append_datum(const Datum& datum). So I cannot use append(datum, N), but can only use append_datum(const Datum& datum).

Comment (silverbullet233) [reply be/src/exprs/array_functions.cpp:1660]: how about append_value_multiple_times? @chenminghua8

**Review by murphyatwork — DISMISSED** [review]

**Review by silverbullet233 — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Review by stdpain — DISMISSED** [review]

**Comment by chenminghua8** [comment]
@silverbullet233 @kangkaisen @stdpain @mofeiatwork My unit tests have covered all the code paths. Why are there many code paths not covered in codecov/project/be-total? Is it calculated based on the code paths covered by the sql test?

**Review by murphyatwork — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wangsimo0** [comment]
ignore backport check: 3.3.1

**Comment by murphyatwork** [comment]
@Mergifyio backport branch-3.3

**Pull request merged**